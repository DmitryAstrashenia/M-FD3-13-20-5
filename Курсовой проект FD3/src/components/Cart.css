.Cart {
    padding: 20px 40px 80px;
}

.Cart .center{
    text-align: center;
}

.Cart .cartView {
    padding: 10px 0;
}

.Cart .cartView li {
    list-style: none;
    margin: 10px 0;
    text-align: center;
    display: flex;
}

.Cart .cartView li:first-child>div {
    border-bottom: 1px solid black;
    font-weight: bold;
}

.Cart .cartView li>div {
    display: flex;
    flex: 3 33.333%;
    justify-content: center;
    align-items: center;
    font-weight: 400;
}

.Cart .cartView p {
    line-height: 1.4;
    margin: 16px 0;
}


.Cart .cartView li:last-child {
    border-top: 1px solid black;
}

.Cart .cartView img {
    width: 25%;
}

.Cart .cartView li>div {
    display: flex;
    flex: 3 33.333%;
    justify-content: center;
    align-items: center;
    font-weight: 400;
}

.Cart .cartView .total {
    text-align: right;
    width: 100%;
    font-weight: bold;
}

.Cart .cartView button {
    width: 30px;
    height: 30px;
    display: inline-block;
    color: black;
    font-size: 90%;
    font-weight: 700;
    text-decoration: none;
    user-select: none;
    padding: .25em .5em;
    outline: none;
    border: 1px solid rgb(250, 172, 17);
    border-radius: 7px;
    background: rgb(255, 212, 3) linear-gradient(rgb(255, 212, 3), rgb(248, 157, 23));
    box-shadow: inset 0 -2px 1px rgba(0, 0, 0, 0), inset 0 1px 2px rgba(0, 0, 0, 0), inset 0 0 0 60px rgba(255, 255, 0, 0);
    transition: box-shadow .2s, border-color .2s;
    margin-right: 20px;
    margin-left: 20px;
}

.Cart .cartView button:hover {
    box-shadow: inset 0 -1px 1px rgba(0, 0, 0, 0), inset 0 1px 2px rgba(0, 0, 0, 0), inset 0 0 0 60px rgba(255, 255, 0, .5);
}

.Cart .cartView button:active {
    padding: calc(.25em + 1px) .5em calc(.25em - 1px);
    border-color: rgba(177, 159, 0, 1);
    box-shadow: inset 0 -1px 1px rgba(0, 0, 0, .1), inset 0 1px 2px rgba(0, 0, 0, .3), inset 0 0 0 60px rgba(255, 255, 0, .45);
}


@media (max-width: 544px) {

    .Cart .cartView button{
        margin-right: 5px;
        margin-left: 10px;
    }

    .Cart {
        padding: 20px 20px 80px;
    }

    .Cart .cartView p{
        font-size: 14px;
    }
    
}


