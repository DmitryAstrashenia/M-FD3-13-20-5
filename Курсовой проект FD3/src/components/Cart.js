"use scrict";

import React, { Fragment } from "react";
import ProductInCart from "./ProductInCart";
import FormOrder from "./FormOrder";
import cartOrderReducer from "./function/cartOrderReducer";
import { createStore } from "redux";
import { myEvents } from "./module/events";
import "./Cart.css";

class Cart extends React.Component {
  store = createStore(cartOrderReducer);
  state = {
    cartOrder: this.store.getState(),
    allGoods: JSON.parse(localStorage.getItem("allGoods")),
  };

  plusItem = (id) => {
    this.store.dispatch({ type: "counter/incremented", id: id });
    localStorage.setItem("cartOrder", JSON.stringify(this.store.getState()));
    this.setState({
      cartOrder: this.store.getState(),
    });
    myEvents.emit("amountGoodsInCart");
  };

  minusItem = (id) => {
    this.store.dispatch({ type: "counter/decremented", id: id });
    if (this.store.getState()[id]) {
      localStorage.setItem("cartOrder", JSON.stringify(this.store.getState()));
      this.setState({
        cartOrder: this.store.getState(),
      });
      myEvents.emit("amountGoodsInCart");
    } else {
      this.deleteItem(id);
      myEvents.emit("amountGoodsInCart");
    }
  };

  deleteItem = (id) => {
    this.store.dispatch({ type: "counter/delete", id: id });
    localStorage.setItem("cartOrder", JSON.stringify(this.store.getState()));
    this.setState({
      cartOrder: this.store.getState(),
    });
    myEvents.emit("amountGoodsInCart");
  };

  componentDidMount = () => {
    myEvents.addListener("plusItem", this.plusItem);
    myEvents.addListener("minusItem", this.minusItem);
    myEvents.addListener("deleteItem", this.deleteItem);
  };

  componentWillUnmount = () => {
    myEvents.removeListener("plusItem", this.plusItem);
    myEvents.removeListener("minusItem", this.minusItem);
    myEvents.removeListener("deleteItem", this.deleteItem);
  };

  render() {
    console.log("Cart-Корзина");
    let products = [];
    let sum = 0;
    if (this.state.cartOrder) {
      let keys = Object.keys(this.state.cartOrder);
      products = keys.map((key) => {
        return (
          <ProductInCart
            key={key}
            id={key}
            allGoods={this.state.allGoods}
            cartOrder={this.state.cartOrder[key]}
          />
        );
      });
      keys.forEach((key) => {
        sum += this.state.allGoods[key]["price"] * this.state.cartOrder[key];
      });
    }

    if (
      this.state.cartOrder === null ||
      !Object.keys(this.state.cartOrder).length > 0
    ) {
      return (
        <div className="Cart">
          <p className="center">Корзина пуста!</p>
          <FormOrder />
        </div>
      );
    } else {
      return (
        <div className="Cart">
          <ul className="cartView">
            <li>
              <div>
                <p>Товар</p>
              </div>
              <div>
                <p>Количество</p>
              </div>
              <div>
                <p>Цена</p>
              </div>
            </li>
            {products}
            <li>
              <p className="total">Итого: {sum} BYN</p>
            </li>
          </ul>
          <FormOrder />
        </div>
      );
    }
  }
}

export default Cart;
