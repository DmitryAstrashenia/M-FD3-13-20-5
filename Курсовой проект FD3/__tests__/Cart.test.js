"use strict";

import React from "react";
import renderer from "react-test-renderer";
import Cart from "../src/components/Cart";

let cardOrder = { 10002: 2 };
let allGoods = {
  10001: {
    articul: "101",
    name: "20C6.2X006 ",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "121",
    pieces: "0",
  },
  10002: {
    articul: "102",
    name: "20C8.2X006",
    img:
      "https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcQECRVdCxETE-6-aAflOsCwbu6pVz35Tt40cg&usqp=CAU",
    price: "125",
    pieces: "5",
  },
  10003: {
    articul: "103",
    name: "20С12X006",
    img:
      "https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcSWCPVCG1nbX23LZ-Vm4G6hUEM2Qnjjcw6tKQ&usqp=CAU",
    price: "180",
    pieces: "7",
  },
  10004: {
    articul: "104",
    name: "20С12X006",
    img:
      "https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcQxRvssQMHmGCy7eg_EtkKc3V59AZ05xXg9JA&usqp=CAU",
    price: "195",
    pieces: "0",
  },
  10005: {
    articul: "105",
    name: "20С12X007",
    img:
      "https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcS0cg5OlK3bFKZhxUVvT5WGS4H8HaRSVF9ICQ&usqp=CAU",
    price: "196",
    pieces: "2",
  },
  10006: {
    articul: "106",
    name: "20С12X008",
    img:
      "https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcStf6OmuyN1Zif_hGDVWvlQoNO0RJXxk8uLQw&usqp=CAU",
    price: "197",
    pieces: "10",
  },
  10007: {
    articul: "107",
    name: "20С12X009",
    img:
      "https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcT9aH7ASep3UscyxbDEFlnUuk73mZD4ec7vDQ&usqp=CAU",
    price: "198",
    pieces: "3",
  },
  10008: {
    articul: "108",
    name: "20С12X010",
    img:
      "https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcTB3-lMQUnQ5ZTcJdDn3-Y-3j9ojcyl59WSJQ&usqp=CAU",
    price: "199",
    pieces: "3",
  },
  10009: {
    articul: "109",
    name: "20С12X011",
    img:
      "https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcTB3-lMQUnQ5ZTcJdDn3-Y-3j9ojcyl59WSJQ&usqp=CAU",
    price: "200",
    pieces: "4",
  },
  10010: {
    articul: "110",
    name: "20С12X012",
    img:
      "https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcT9aH7ASep3UscyxbDEFlnUuk73mZD4ec7vDQ&usqp=CAU",
    price: "201",
    pieces: "8",
  },
  10011: {
    articul: "111",
    name: "20С12X013",
    img:
      "https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcStf6OmuyN1Zif_hGDVWvlQoNO0RJXxk8uLQw&usqp=CAU",
    price: "202",
    pieces: "0",
  },
  10012: {
    articul: "112",
    name: "20С12X014",
    img:
      "https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcQxRvssQMHmGCy7eg_EtkKc3V59AZ05xXg9JA&usqp=CAU",
    price: "203",
    pieces: "0",
  },
  10013: {
    articul: "113",
    name: "20С12X015",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "204",
    pieces: "1",
  },
  10014: {
    articul: "114",
    name: "20С12X016",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "205",
    pieces: "1",
  },
  10015: {
    articul: "115",
    name: "20С12X017",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "206",
    pieces: "2",
  },
  10016: {
    articul: "116",
    name: "20С12X018",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "207",
    pieces: "3",
  },
  10017: {
    articul: "117",
    name: "20С12X019",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "208",
    pieces: "5",
  },
  10018: {
    articul: "118",
    name: "20С12X020",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "209",
    pieces: "4",
  },
  10019: {
    articul: "119",
    name: "20С12X021",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "210",
    pieces: "5",
  },
  10020: {
    articul: "120",
    name: "20С12X022",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "211",
    pieces: "5",
  },
  10021: {
    articul: "121",
    name: "20С12X023",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "212",
    pieces: "5",
  },
  10022: {
    articul: "122",
    name: "20С12X024",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "213",
    pieces: "5",
  },
  10023: {
    articul: "123",
    name: "20С12X025",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "214",
    pieces: "5",
  },
  10024: {
    articul: "124",
    name: "20С12X026",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "215",
    pieces: "5",
  },
  10025: {
    articul: "125",
    name: "20С12X027",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "216",
    pieces: "5",
  },
  10026: {
    articul: "126",
    name: "20С12X028",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "217",
    pieces: "5",
  },
  10027: {
    articul: "127",
    name: "20С12X029",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "218",
    pieces: "5",
  },
  10028: {
    articul: "128",
    name: "20С12X030",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "219",
    pieces: "5",
  },
  10029: {
    articul: "129",
    name: "20С12X031",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "220",
    pieces: "5",
  },
  10030: {
    articul: "130",
    name: "20С12X032",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "221",
    pieces: "5",
  },
  10031: {
    articul: "131",
    name: "20С12X033",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "222",
    pieces: "5",
  },
  10032: {
    articul: "132",
    name: "20С12X034",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "223",
    pieces: "5",
  },
  10033: {
    articul: "133",
    name: "20С12X035",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "224",
    pieces: "5",
  },
  10034: {
    articul: "134",
    name: "20С12X036",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "225",
    pieces: "5",
  },
  10035: {
    articul: "135",
    name: "20С12X037",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "226",
    pieces: "5",
  },
  10036: {
    articul: "136",
    name: "20С12X038",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "227",
    pieces: "5",
  },
  10037: {
    articul: "137",
    name: "20С12X039",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "228",
    pieces: "5",
  },
  10038: {
    articul: "138",
    name: "20С12X040",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "229",
    pieces: "5",
  },
  10039: {
    articul: "139",
    name: "20С12X041",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "230",
    pieces: "5",
  },
  10040: {
    articul: "140",
    name: "20С12X042",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "231",
    pieces: "5",
  },
  10041: {
    articul: "141",
    name: "20С12X043",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "232",
    pieces: "5",
  },
  10042: {
    articul: "142",
    name: "20С12X044",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "233",
    pieces: "5",
  },
  10043: {
    articul: "143",
    name: "20С12X045",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "234",
    pieces: "5",
  },
  10044: {
    articul: "144",
    name: "20С12X046",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "235",
    pieces: "5",
  },
  10045: {
    articul: "145",
    name: "20С12X047",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "236",
    pieces: "5",
  },
  10046: {
    articul: "146",
    name: "20С12X048",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "237",
    pieces: "5",
  },
  10047: {
    articul: "147",
    name: "20С12X049",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "238",
    pieces: "5",
  },
  10048: {
    articul: "148",
    name: "20С12X050",
    img:
      "https://dimon713.ru/src/img/gear_pump_hydra_pack/Bezimyannij-150x150.png",
    price: "239",
    pieces: "5",
  },
  20001: {
    articul: "201",
    name: "20C6.2X006 ",
    img:
      "https://dimon713.ru//src/img/pump_motor/828917111_w640_h640_a2fo_br6x_main__to_rgb.big-150x150.jpg",
    price: "120",
    pieces: "0",
  },
  20002: {
    articul: "202",
    name: "20C8.2X006",
    img:
      "https://dimon713.ru//src/img/pump_motor/828917111_w640_h640_a2fo_br6x_main__to_rgb.big-150x150.jpg",
    price: "125",
    pieces: "4",
  },
  20003: {
    articul: "203",
    name: "20C10X006",
    img:
      "https://dimon713.ru//src/img/pump_motor/828917111_w640_h640_a2fo_br6x_main__to_rgb.big-150x150.jpg",
    price: "170",
    pieces: "2",
  },
  20004: {
    articul: "204",
    name: "20C12X006",
    img:
      "https://dimon713.ru//src/img/pump_motor/828917111_w640_h640_a2fo_br6x_main__to_rgb.big-150x150.jpg",
    price: "175",
    pieces: "6",
  },
};

localStorage.setItem("cartOrder", JSON.stringify(cardOrder));
localStorage.setItem("allGoods", JSON.stringify(allGoods));

test("работа c корзиной/Cart", () => {
  const component = renderer.create(<Cart />);

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();

  const buttonElem = component.root.findAll((el) => el.type == "button");

  buttonElem[0].props.onClick();
  expect(JSON.parse(localStorage.getItem("cartOrder"))).toEqual({ 10002: 1 });

  buttonElem[1].props.onClick();
  expect(JSON.parse(localStorage.getItem("cartOrder"))).toEqual({ 10002: 2 });

  buttonElem[2].props.onClick();
  expect(JSON.parse(localStorage.getItem("cartOrder"))).toEqual({});

  componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});
