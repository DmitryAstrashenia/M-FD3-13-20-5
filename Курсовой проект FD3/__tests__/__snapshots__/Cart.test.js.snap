// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа c корзиной/Cart 1`] = `
<div
  className="Cart"
>
  <ul
    className="cartView"
  >
    <li>
      <div>
        <p>
          Товар
        </p>
      </div>
      <div>
        <p>
          Количество
        </p>
      </div>
      <div>
        <p>
          Цена
        </p>
      </div>
    </li>
    <li>
      <div>
        <img
          src="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcQECRVdCxETE-6-aAflOsCwbu6pVz35Tt40cg&usqp=CAU"
        />
        <p>
          20C8.2X006
        </p>
      </div>
      <div>
        <button
          className="minus"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
        <p>
          2
          шт.
        </p>
        <button
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div>
        <p>
          250
          BYN
        </p>
        <button
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </div>
    </li>
    <li>
      <p
        className="total"
      >
        Итого: 
        250
         BYN
      </p>
    </li>
  </ul>
  <div
    className="wpapper_feedback"
  >
    <h2
      className="center"
    >
      Форма заказа
    </h2>
    <p
      className="center"
    >
      Для заказа заполните форму и нажмите отправить:
    </p>
    <form
      className="feedback"
      id="form_cart"
      onSubmit={[Function]}
    >
      <div
        className="feedback-group "
      >
        <label
          className="feedback-label"
        >
          <span>
            Имя:
          </span>
          <input
            className="feedback-input"
            id="user"
            onChange={[Function]}
            placeholder="Введите Ваше имя"
            type="text"
          />
          <div
            className="err-message userErr invisible"
          >
            Введите ФИО
          </div>
          <i
            className="far fa-user"
          />
        </label>
        <label
          className="feedback-label"
        >
          <span>
            Email:
          </span>
          <input
            className="feedback-input"
            id="email"
            onChange={[Function]}
            placeholder="Введите адрес Вашей почты"
            type="email"
          />
          <div
            className="err-message emailErr invisible"
          >
            Введите почту для связи
          </div>
          <i
            className="far fa-envelope"
          />
        </label>
        <label
          className="feedback-label"
        >
          <span>
            Телефон:
          </span>
          <input
            className="feedback-input"
            id="tel"
            onChange={[Function]}
            placeholder="Введите номер телефона"
            type="tel"
          />
          <div
            className="err-message telErr invisible"
          >
            Введите номер телефона
          </div>
          <i
            className="fas fa-mobile-alt"
          />
        </label>
        <label
          className="feedback-label"
        >
          <span>
            Комментарий:
          </span>
          <textarea
            className="feedback-textarea"
            id="customer-text"
            placeholder="Здравствуйте, БПИ!"
          />
          <i
            className="fas fa-pen-alt"
          />
        </label>
        <button
          className="feedback-button"
          onClick={[Function]}
          type="button"
        >
          Отправить
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`работа c корзиной/Cart 2`] = `
<div
  className="Cart"
>
  <p
    className="center"
  >
    Корзина пуста!
  </p>
  <div
    className="wpapper_feedback"
  >
    <h2
      className="center"
    >
      Форма заказа
    </h2>
    <p
      className="center"
    >
      Для заказа заполните форму и нажмите отправить:
    </p>
    <form
      className="feedback"
      id="form_cart"
      onSubmit={[Function]}
    >
      <div
        className="feedback-group "
      >
        <label
          className="feedback-label"
        >
          <span>
            Имя:
          </span>
          <input
            className="feedback-input"
            id="user"
            onChange={[Function]}
            placeholder="Введите Ваше имя"
            type="text"
          />
          <div
            className="err-message userErr invisible"
          >
            Введите ФИО
          </div>
          <i
            className="far fa-user"
          />
        </label>
        <label
          className="feedback-label"
        >
          <span>
            Email:
          </span>
          <input
            className="feedback-input"
            id="email"
            onChange={[Function]}
            placeholder="Введите адрес Вашей почты"
            type="email"
          />
          <div
            className="err-message emailErr invisible"
          >
            Введите почту для связи
          </div>
          <i
            className="far fa-envelope"
          />
        </label>
        <label
          className="feedback-label"
        >
          <span>
            Телефон:
          </span>
          <input
            className="feedback-input"
            id="tel"
            onChange={[Function]}
            placeholder="Введите номер телефона"
            type="tel"
          />
          <div
            className="err-message telErr invisible"
          >
            Введите номер телефона
          </div>
          <i
            className="fas fa-mobile-alt"
          />
        </label>
        <label
          className="feedback-label"
        >
          <span>
            Комментарий:
          </span>
          <textarea
            className="feedback-textarea"
            id="customer-text"
            placeholder="Здравствуйте, БПИ!"
          />
          <i
            className="fas fa-pen-alt"
          />
        </label>
        <button
          className="feedback-button"
          onClick={[Function]}
          type="button"
        >
          Отправить
        </button>
      </div>
    </form>
  </div>
</div>
`;
